airflow/providers/discord/notifications/discord.py

```diff
@@ -0,0 +1,81 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from airflow.compat.functools import cached_property
+from airflow.exceptions import AirflowOptionalProviderFeatureException
+
+try:
+    from airflow.notifications.basenotifier import BaseNotifier
+except ImportError:
+    raise AirflowOptionalProviderFeatureException(
+        "Failed to import BaseNotifier. This feature is only available in Airflow versions >= 2.6.0"
+    )
+
+from airflow.providers.discord.hooks.discord_webhook import DiscordWebhookHook
+
+ICON_URL: str = "https://raw.githubusercontent.com/apache/airflow/2.5.0/airflow/www/static/pin_100.png"
+
+
+class DiscordNotifier(BaseNotifier):
+    """
+    Discord BaseNotifier
+
+    :param discord_conn_id: Http connection ID with host as "https://discord.com/api/" and
+                         default webhook endpoint in the extra field in the form of
+                         {"webhook_endpoint": "webhooks/{webhook.id}/{webhook.token}"}
+    :param text: The content of the message
+    :param username: The username to send the message as. Optional
+    :param avatar_url: The URL of the avatar to use for the message. Optional
+    :param tts: Text to speech.
+    """
+
+    # A property that specifies the attributes that can be templated.
+    template_fields = ("discord_conn_id", "text", "username", "avatar_url", "tts")
+
+    def __init__(
+        self,
+        discord_conn_id: str = "discord_webhook_default",
+        text: str = "This is a default message",
+        username: str = "Airflow",
+        avatar_url: str = ICON_URL,
+        tts: bool = False,
+    ):
+        super().__init__()
+        self.discord_conn_id = discord_conn_id
+        self.text = text
+        self.username = username
+        self.avatar_url = avatar_url
+
+        # If you're havin problem with tts not recognized in __init__
+        # you can define that after instantiate the class
```

Comment:
```suggestion
        # you can define that after instantiating the class
```

Reply:
Also could you please elaborate on what could be the potential problem here? If the argument is not recognisable does it allow to instantiate the class or does it raise an exception? If an exception does this need to be caught?